src/Symfony/Component/Translation/Tests/Dumper/XliffFileDumperTest.php

```diff
@@ -38,4 +38,33 @@ public function testDump()
 
         unlink($tempDir.'/messages.en_US.xlf');
     }
+
+    public function testExtendedDumperCanSetCustomToolInfo()
+    {
+        $dumper = new XliffExtendedFileDumper();
+
+        $this->assertContains(
+            '<tool tool-id="symfony" tool-name="Test" tool-version="0.0"/>',
+            $dumper->formatTestMessages( array() )
+        );
+    }
+
 }
+
+/**
+ * Test subclass for configuring dumper via protected functions.
+ */
+class XliffExtendedFileDumper extends XliffFileDumper {
+    public function formatTestMessages( array $messages ){
```

Comment:
Any spaces I added were to emulate the style of the rest of the file, rather than my personal choice. I read in the community guidelines this was the expected form.

The fabbot sent me a patch on my first PR attempt and I duly applied it.